### ISSUE TITLE
[Bug]: No rewards are being given out?

### ISSUE THREAD

**Issue by ILikeOrangeJuice** [Oheers/EvenMoreFish#381]
### Contact Details

Discord GaZZip 

### What happened?

Today we updated from 1.20.4 to the latest 1.20.6 paper and updated emf to the latest version on spigot and ran our competitions like usual, we made no changess to the plugig or config or anything regarding it.
The competition and everything works flawless, however, the rewards are not being sent out.
I now tried the latest dev release from your jenkins, however unfortunately it was the same result.

### Version

Development

### Development Version

1.7.2#73

### Server Version

1.20.6#148

### Java Version

21

### Relevant log output

```shell
Unfortunately there is no error or anything, it sends out who wins, but just the rewards do not fire. I tested the commands manually from the plugin we use (crates) and it worked, so it's not on the plugins side.

Skyblock competitions.yml https://pastebin.com/0WUNaAeP
Citybuild competitions.yml https://pastebin.com/BxH3ydri

Citybuild emf related log entries after starting the development build: https://pastebin.com/AySTJJSs
Citybuild all entries for the 19:00 competition with the dev version: https://pastebin.com/TTAv2mwu

Again, no rewards given out.
```

### Relevant screenshots

Ingame screenshot of /emf admin version from Skyblock https://prnt.sc/z68OAyPooSbH
Ingame screenshot of /emf admin version from Citybuild  https://prnt.sc/CTbncKMixktb
(German) but here you can see the winners, none received anything tho. https://prnt.sc/b0MVuN6sFt9f
Citybuild 19:00 beta version competition, winners announced but no rewards again https://prnt.sc/xCCSjTgg10eC

**Comment by ILikeOrangeJuice** [comment]
Using the dev build #75 did not fix it unfortunately., we run multiple competitions a day, this is pure chaos haha.

**Comment by FireML** [comment]
does this show up in your console at all?
`An exception was thrown while the competition was being ended!`

The competitions were changed to warn console instead of breaking themselves if an exception is thrown.

**Comment by sarhatabaot** [comment]
Can you post your rewards setup? If you don't want to send it here, you can dm me on discord (sarhatabaot)

**Comment by ILikeOrangeJuice** [comment]
Hey, I even tried a fresh install on a 1.20.6 testserver of us, just changing the date and a broadcast command added to the rewards of it, however nothing came up. @FireML Nope, just nothing happening.
This is the current competitions setup we use and it worked as said flawless in 1.20.4.
Citybuild: 
https://pastebin.com/0kTq6bC0
Skyblock:
https://pastebin.com/BhQ5SLEC

Again, everything but the rewards work flawless, any command, any timer, all, or we did not have any other issue at least haha.
Only the rewards don't seem to trigger like at all. I even tried different types in case it may be one of my used ones, but even a clear fresh install on an basically fresh test server has the same result unfortunately.

Here are also my versions again:
Emf 1.7.2#75, currently the latest dev release.
Paper 1.20.6#148
Protocollib; 5.3.0-SNAPSHOT-723

**Comment by ILikeOrangeJuice** [comment]
I tested version 1.7 released on May 8, 2024 on SpigotMC and it seems to have worked to give out rewards on our testserver. Meaning the issue must rely somewhere between 1.7.1 and the latest dev or 1.7.2. I hope that helps, this is still urgent.

**Comment by ILikeOrangeJuice** [comment]
I went even further and tested more versions... 1.7.1 seems to work too, it started not working with 1.7.2, so that one is the culprit.

**Comment by FireML** [comment]
I forgot to mention, but this was fixed in #429
